web/js/codeworld_shared.js

```diff
@@ -1559,78 +1559,101 @@ function preFormatMessage(msg) {
 }
 
 function printMessage(type, message) {
-  const outputDiv = document.getElementById('message');
+  const $outputBlock = $('#message');
+  const $lastOutputBlock = $outputBlock.children().last();
 
-  let box = outputDiv.lastChild;
-  let messageContent;
-  if (box && type === 'log' && box.classList.contains('log')) {
-    box.rawMessage += message;
-    messageContent = box.lastChild;
-  } else {
-    box = document.createElement('div');
-    box.classList.add('message-box');
-    box.classList.add(type);
+  const formattedMessage = preFormatMessage(message);
+  const lines = formattedMessage.trim().split('\n');
 
-    const messageGutter = document.createElement('div');
-    messageGutter.classList.add('message-gutter');
+  // Combine sequential log messages.
+  if (
+    type === 'log' &&
+    $lastOutputBlock.length &&
+    $lastOutputBlock.attr('class').includes('log')
+  ) {
+    const $messageContent = $lastOutputBlock.find('.message-content');
 
-    messageContent = document.createElement('div');
-    messageContent.classList.add('message-content');
+    if ($messageContent.children().length) {
+      $messageContent.append(lines);
+    } else {
```

Comment:
Will this branch ever get taken?  You're always adding at least one child to any element.

Reply:
Yes, it will be in case of a single-line message as it contains only text:

```html
<div>
  {lines}
</div>
```

On the other hand, multiline messages always contain a `summary` tag, and that's where we'd like to inject the report link:

```html
<details>
  <summary>
    {first_line}
  </summary>
  {remaining_lines}
</details>
```